src/Producer/RecordValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+
+namespace Kafka\Producer;
+
+use Kafka\Broker;
+use Kafka\Exception;
+use function is_string;
+use function trim;
+
+class RecordValidator
```

Comment:
This should be `final`

Reply:
And we should cover this class with unit tests

Reply:
I found difficult mocking final classes without interface, that's why I didn't do that. Usually, I don't mock but the guy I work on project with does and hates finals then.  What's your opinion on this?

Reply:
In general lines (and in a very naive way): mock only abstractions that you own 😃

Reply:
The main issue here is that we're extracting a component which is extremely coupled to the sync/async process, mostly because we're doing some array voodoo.

In that sense I don't think that we should mock that class.

Reply:
We should even mark that class as `@internal` to clarify that nobody should ever depend on it (ahh private classes - ❤️ Java)

Reply:
In Java I even mark all the parameters and internal variables as `final`, the code gets **so much** simpler.